packages/pages/src/common/src/assets/getAssetsFilepath.ts

```diff
@@ -0,0 +1,39 @@
+import fs from "node:fs";
+import { pathToFileURL } from "url";
+import { UserConfig } from "vite";
+import { import_ } from "./import.js";
+import yaml from "js-yaml";
+import { ConfigYaml } from "../config/config.js";
+
+/**
+ * Determines the assets directory to use by checking the following, in order:
+ * 1. config.yaml
+ * 2. vite.config.json assetDir
+ * 3. default to "assets"
+ * @param servingJsonPath the path to sites-config/serving.json - make sure to take scope into account
+ * @param viteConfigPath the path to vite.config.js
+ */
+export const determineAssetsFilepath = async (
+  defaultAssetsDir: string,
+  configYamlPath: string,
+  viteConfigPath: string
+) => {
+  if (configYamlPath === "" || viteConfigPath === "") {
+    return defaultAssetsDir;
+  }
+
+  if (fs.existsSync(configYamlPath)) {
```

Comment:
I feel like the logic to read the config.yaml file should happen elsewhere and shouldn't be specific to asset filepath determination.

Other places need to read it anyways right?

Reply:
That's a good callout. I had the config.yaml stuff as a TODO but decided to add it in here now since I already wrote the code and tests for reading a file. I will refactor this in an upcoming PR when we introduce the true consolidated config work. Going to leave here for now though.